modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -1615,7 +1642,19 @@ def test_reset_index_with_named_index(
     df_equals(modin_df.reset_index(drop=False), pandas_df.reset_index(drop=False))
 
 
-@pytest.mark.parametrize("test_async_reset_index", [False, True])
+@pytest.mark.parametrize(
+    "test_async_reset_index",
+    [
+        pytest.param(False),
+        pytest.param(
+            True,
+            marks=pytest.mark.skipif(
+                StorageFormat.get() == "Hdk",
+                reason="_propagate_index_objs is not supported by HDK",
```

Comment:
I believe that direct `set_index` usage should help to avoid `_propagate_index_objs` usage.